src/AddOn.Episerver.Settings/Core/SettingsInitializationModule.cs

```diff
@@ -216,23 +215,36 @@ private static void InitCompleteHandler(object sender, EventArgs e)
     /// <param name="e">The <see cref="ContentEventArgs" /> instance containing the event data.</param>
     private static void MovingContent(object sender, ContentEventArgs e)
     {
-        if (e == null)
+        if (e == null || e.Content == null)
         {
             return;
         }
-
-        // if the content item is an instance of SettingsBase, it has been instantiated with the fallback base class
-        // since the ContentType class no longer exists and should be possible to move to the waste basket
-        if (e.Content.GetOriginalType() == typeof(SettingsBase) ||
-            !(e.Content is SettingsBase) ||
-            e.Content.ParentLink != settingsService.GlobalSettingsRoot ||
-            e.TargetLink.ID != 2)
+        
+        // if trying to move something into the global settings folder
+        if (e.TargetLink == settingsService.GlobalSettingsRoot)
         {
+            e.CancelAction = true;
+            e.CancelReason = localizationService.GetString("/edit/movesetting/contenttoglobalsetting");
+
             return;
         }
+        
+        // if it's a global setting
+        if (e.Content is SettingsBase && e.Content.ParentLink == settingsService.GlobalSettingsRoot )
+        {
+            // if the content item is an instance of SettingsBase, it has been instantiated with the fallback base class
+            // since the ContentType class no longer exists and should be possible to move to the waste basket
+            if (e.Content.GetOriginalType() == typeof(SettingsBase) && e.TargetLink == ContentReference.WasteBasket)
+            {
+                return;
+            }
 
-        e.CancelAction = true;
-        e.CancelReason = localizationService.GetString("/edit/deletesetting/deletenotsupported");
+            if (e.TargetLink == ContentReference.WasteBasket)
+            {
+                e.CancelAction = true;
+                e.CancelReason = localizationService.GetString("/edit/deletesetting/deletenotsupported");
+            }
```

Comment:
Is there a case where a settings can be moved outside the settings rootfolders but not the wastebasket that we should prevent?

Reply:
I tried but could not find any way that it was possible to do that action in the UI